app/scenes/Document/components/EditableTitle.tsx

```diff
@@ -50,6 +57,29 @@ const EditableTitle = React.forwardRef(
     }: Props,
     ref: React.RefObject<RefHandle>
   ) => {
+    const theme = useTheme();
+    const pickerTheme = usePickerTheme();
+    const isMobile = useMobile();
+    const activeElement = useActiveElement();
+
+    const [isFocused, setFocus] = React.useState<boolean>(false);
+
+    React.useEffect(() => {
+      if (
+        activeElement &&
+        (["emoji-picker-disclosure", "emoji-picker", "document-title"].includes(
```

Comment:
The `<Title` component exposes ref, so we should be able to use that rather than a hardcoded `id` here for `document-title`?

Reply:
Yes, specifically a [`RefHandle`](https://github.com/outline/outline/blob/main/app/components/ContentEditable.tsx#L57-L79) which is a little problematic since `ref` in this case doesn't seem to expose the `HTMLElement` :end: 

Anyways, one way to tackle/improve this is to have three refs, i.e, `emojiPickerRef`, `disclosureRef` and `titleRef`, and snap these to the corresponding components. Then we could compare these against the `activeElement`. What do you think?

Reply:
TIL [`useId`](https://beta.reactjs.org/apis/react/useId). Maybe we can use this as an alternate?

Reply:
I don't think `useId` really offers anything here. 

> in this case doesn't seem to expose the HTMLElement

Oh true, you'd just want to add it on `useImperativeHandle` in `ContentEditable` as `element` I suppose, that seems like a useful edition.

Reply:
> I don't think `useId` really offers anything here.

True. I was implying using it in the components which form the _focus group_ in our case and then use `ref.current.id === activeDocument.id` for comparison but now I realize that this isn't needed and just `ref.current === activeDocument` would be sufficient.

> you'd just want to add it on useImperativeHandle in ContentEditable as element I suppose

Yes, tried attempting this only to run into issues. Tried :point_down: 

```js
element: () => {
  return contentRef.current;
}
```

only to get the entire custom object passed to `useImperativeHandle` :smile: 

Maybe I did something stupid, but you're right and it occurred to me as well that patching `useImperativeHandle` makes sense. I'll give it a shot again today :+1: